### DISCUSSION TITLE
Return ID when insert row in database

### DISCUSSION THREAD

**Discussion by laldevfr** [serverpod/serverpod#574] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Usually in my app when i add a row i just return to the front the id of item i add 

**Describe the solution you'd like**
The [Insert fonction](https://docs.serverpod.dev/concepts/database-communication#inserting-a-table-row) return the id of the row created

**Comment by vlidholt** [comment, marked as answer]
The id is accessible through the object that you added. E.g.

```dart
await MyClass.insert(myObject);
var insertId = myObject.id!;
```

**Reply by laldevfr** [reply]
Thank's for your answer, but does it work if myObject it's a param ? 
like : 
```dart 
 Future<int?> addTextContenu(Session session,Contenu contenu) async {
    int? userid=await session.auth.authenticatedUserId;
    if (userid!=null) {
      Contenu.insert(session, contenu);
     var insertId = contenu.id!;
      return insertId;
    } else {
      return 0;
    }
  }
```

**Reply by yahu1031** [reply]
Yes, it does work but make sure to return `-1` on else condition.

**Comment by wstrange** [comment]
I think many users will stumble on this expecting the insert call to return the new id, or to return a new copy of the inserted object. 

Could be addressed via documentation - but it feels like the insert api should also return the id.
